include/api/CDataFrameAnalysisInstrumentation.h

```diff
@@ -103,16 +106,19 @@ class API_EXPORT CDataFrameAnalysisInstrumentation
     //! Start polling and writing progress updates.
     //!
     //! \note This doesn't return until instrumentation.setToFinished() is called.
-    static void monitor(const CDataFrameAnalysisInstrumentation& instrumentation,
+    static void monitor(CDataFrameAnalysisInstrumentation& instrumentation,
                         core::CRapidJsonConcurrentLineWriter& writer);
 
 protected:
     using TWriter = core::CRapidJsonConcurrentLineWriter;
     using TWriterUPtr = std::unique_ptr<TWriter>;
+    using TOptionalInt64 = boost::optional<std::int64_t>;
 
 protected:
     virtual counter_t::ECounterTypes memoryCounterType() = 0;
     TWriter* writer();
+    void advisedMemoryLimit(std::int64_t advisedMemoryLimit);
```

Comment:
I'm not sure about calling this `advisedMemoryLimit` it feels a bit vague about what purpose it serves, especially since we have estimated upfront memory usage. I think this should capture something about the fact it relates to out-of-memory and so is an increase, the best I came up with was `suggestedMemoryIncrease`.

Reply:
Renamed in `increasedMemoryEstimate`